eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -321,89 +319,69 @@ jobs:
-        inputs:
-          artifactName: packages
-          path: $(Build.ArtifactStagingDirectory)/packages
-
-      # Generate Documentation
-      - pwsh: |
-          $setupFile="doxygen-1.8.18-setup.exe";
-
-          Invoke-WebRequest `
-            -MaximumRetryCount 10 `
-            -Uri "https://azuresdktooling.blob.core.windows.net/dependencies/$setupFile" `
-            -OutFile $setupFile `
-          | Wait-Process;
-
-          Start-Process -Filepath .\$setupFile -ArgumentList @("/VERYSILENT") -Wait
-        workingDirectory: $(Agent.TempDirectory)
-        displayName: Download and Install Doxygen
-
-      - ${{ each artifact in parameters.Artifacts }}:
-        - pwsh: Write-Host "##vso[task.setvariable variable=PackageVersion]$(Get-Content .\sdk\${{ parameters.ServiceDirectory }}\${{ artifact.Path }}\version.txt)"
-          displayName: Set $(PackageVersion) from version.txt file - ${{ artifact.Name }}
-        - pwsh: >-
-            python eng/docs/api/generate_docs.py
-            --TemplateFile="eng/docs/api/Doxyfile.template"
-            --PackagePath="sdk/${{ parameters.ServiceDirectory }}/${{ artifact.Path }}"
-            --PackageName="${{ artifact.Name }}"
-            --PackageVersion="$(PackageVersion)"
-            --AssetsPath="eng/docs/api/assets"
-            --DoxygenPath="c:\\program files\\doxygen\\bin\\doxygen.exe"
-          displayName: Generate docs - ${{ artifact.Name }}
-
-        - pwsh: New-Item -ItemType directory -Path $(Build.ArtifactStagingDirectory) -Name docs/${{ artifact.Name }}
-          displayName: Create doc artifact location - ${{ artifact.Name }}
-
-        - task: CopyFiles@2
-          inputs:
-            sourceFolder: sdk/${{ parameters.ServiceDirectory }}/${{ artifact.Path }}/html
-            targetFolder: $(Build.ArtifactStagingDirectory)/docs/${{ artifact.Name }}
-          displayName: Copy documentation - ${{ artifact.Name }}
-
-        - pwsh: |
-            $packageInfoPath = Join-Path -Path $(Build.ArtifactStagingDirectory) packages/${{ artifact.Name }}/package-info.json
-            Copy-Item -Path $packageInfoPath -Destination $(Build.ArtifactStagingDirectory)/docs/${{ artifact.Name }}/package-info.json
-          displayName: Copy package-info.json to documentation path
-
-      - task: PublishPipelineArtifact@1
-        inputs:
-          artifactName: docs
-          path: $(Build.ArtifactStagingDirectory)/docs
+
+- job: GenerateReleaseArtifacts
+  pool:
+    vmImage: windows-2019
+  steps:
+    # Generate release-info
+    - pwsh: >-
+        New-Item
+        -ItemType directory
+        -Path $(Build.ArtifactStagingDirectory)
+        -Name release-info
+      displayName: Create package info artifact location
+
+    - pwsh: |
+        $outputPath = Join-Path -Path $(Build.ArtifactStagingDirectory) release-info/package-info.json
+        $VersionStringRegEx = '#define AZ_SDK_VERSION_STRING "(([0-9]+)\.([0-9]+)\.([0-9]+)(\-[^\"\-]+)?)"';
```

Comment:
I suggest not having the number regex here but a simple version that just pulls out the data. If we want to validate then we should use the SemVer library.

Reply:
Simplified and refactored to less copies of this.